### PR TITLE
refactor onnxslim test and fix bugs

### PR DESCRIPTION
**Pull request by whyb**
Closes https://github.com/inisis/OnnxSlim/issues/42

Added OnnxSlim command test.

* Test the basic functionality of the slim function
* Test the modification of input shapes.
* Test the conversion of an ONNX model from FP32 to FP16 precision.
* Test the conversion of an ONNX model from FP16 to FP32 precision.

### CODE REVIEW

**Comment by inisis** [comment]
Good Job!

**Comment by whyb** [comment]
@inisis 

Could not merge this pr to main branch, Because CI test failed.

Test command:
```shell
onnxslim --input-shapes input:1,3,224,224 "/data/modelzoo/resnet18/resnet18.onnx" "/tmp/tmpj1kh/resnet18_fp16.onnx“
```

Failed message:
```shell
 =========================== short test summary info ============================
FAILED tests/test_onnxslim.py::TestFunctional::test_input_shape_modification - assert 2 == 0
 +  where 2 = CompletedProcess(args='onnxslim --input-shapes input:1,3,224,224 "/data/modelzoo/resnet18/resnet18.onnx" "/tmp/tmpj1kh...-v]\n                input_model [output_model]\nonnxslim: error: the following arguments are required: input_model\n').returncode
FAILED tests/test_onnxslim.py::TestFunctional::test_fp322fp16_conversion - FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpgmnl41ah/resnet18_fp16.onnx'
================== 2 failed, 5 passed, 107 warnings in 10.90s ==================
```

File: tests/test_onnxslim.py [hunk 1]
```diff
@@ -12,21 +12,36 @@
...
             print_model_info_as_table(summary)
-            output_name = os.path.join(tempdir, "resnet18.onnx")
-            slim(FILENAME, output_name)
-            slim(FILENAME, output_name, model_check=True)
-
-            command = f"onnxslim {FILENAME} {output_name}"
+            out_model_path = os.path.join(tempdir, out_model_name)
+            slim(in_model_path, out_model_path)
+            slim(in_model_path, out_model_path, model_check=True)
+            command = f'onnxslim {arg_str} "{in_model_path}" "{out_model_path}"'
```
Comment (inisis) [inline tests/test_onnxslim.py:22]: I think the *arg_str* should be placed after *input_model* and *output_model*

**Review by inisis — CHANGES_REQUESTED** [review]

**Review by inisis — APPROVED** [review]

**Review by inisis — APPROVED** [review]

**Comment by inisis** [comment]
@whyb Thanks for your contributions!

**Pull request merged**